lib/include/jxl/codestream_header.h

```diff
@@ -229,6 +229,11 @@ typedef struct JxlBasicInfo {
    * used if have_animation is JXL_TRUE.
    */
   JxlAnimationHeader animation;
+
+  /** Padding for forwards-compatibility, in case more fields are exposed
+   * in a future version of the library.
+   */
+  uint8_t padding[108];
```

Comment:
It would be good idea to have a static_assert in libjxl code (probably decode.cc) to check that the size of this struct is a given number, and explain that this value should not be changed, etc

Reply:
Added a static_assert and documented changes in the changelog.

Reply:
JXL bool is typedeffed as "int", which does not have a well defined size. Should we change it to something else, like int32_t? char and even int8_t may be less preferable due to how C deals with chars

Reply:
Can we redefine JXL_BOOL to be a well-defined type?

Maybe int32 is a bit wasteful, not that it probably matters much. What's the issue with using int8 or uint8?

Reply:
Let's ensure that our non-opaque structs have well defined sizes in a follow-up pull request.